src/middleware/dataset.ts

```diff
@@ -14,8 +15,7 @@ export type DatasetContext<Context extends DefaultContextExtends = DefaultContex
   DataFactoryContext<Context & { request: WithDataset<Request, Q>; response: WithDataset<Response, Q> },
   ExtendedDataFactory<Q>>;
 
-export default (): Middleware<DefaultStateExtends, DataFactoryContext<ExtendableContext,
-ExtendedDataFactory<BaseQuad>>> => (
+export default (): Middleware<DataFactoryContext<ExtendableContext, ExtendedDataFactory<BaseQuad>>> => (
```

Comment:
ah, this it not the same `Middleware` as before which explains the difference in the generic types